google-cloud-bigtable/lib/google/cloud/bigtable/rows_reader.rb

```diff
@@ -137,19 +152,23 @@ def retry_options rows_limit, row_set
               end
             end
 
-            delete_indexes.each { |i| row_set.row_ranges.delete_at i }
+            delete_indexes.sort.reverse_each { |i| row_set.row_ranges.delete_at i }
```

Comment:
why do you need a sort here?

Reply:
Here we delete by index while updating the set. Sorting helps us deletes from the end instead of from the beginning, which would mess up the index we want to delete

Reply:
But the indices are sorted already? that array is populated from `.each_with_index` which will generate indices in ascending order?

Having said that, I think what you want is:
```
row_set.row_ranges.reject! {|r| end_key_read? r }

```

Reply:
Also I think we are missing the bit where the last_key happens to be in the middle of a range?

Reply:
oops github folded over the elseif :(